apps/playground/src/App.tsx

```diff
@@ -40,6 +56,41 @@ function App() {
     fileModels().map((model) => model.uri.path),
   );
 
+  loadSavedModels().then(setSavedModels).catch(console.error);
+  setTimeout(() => setSavedModels([]), 5000);
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  const saveModelsDebounced = debounce(() => saveModels(models()), 1000);
+
+  createEffect(() => {
+    if (models().some((m) => m.getValueLength() > 0)) {
+      saveModelsDebounced();
+    }
+  });
+
+  function restoreSavedModels() {
+    batch(() => {
+      models().forEach((model) => model.dispose());
```

Comment:
> Does this clear any currently open tabs

Yes, they can't be kept because each model needs to have an unique URI (path)
A possibility would be saving them before or showing a confirm dialog